File: 150_evaluate_reverse_polish_notation/evalRPN.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if len(token) == 1 and ord(token) < 48:
                b = stack.pop()
                a = stack.pop()
                operator = token
                res = self.operations(a, b, operator)
                stack.append(res)
            else:
                stack.append(int(token))
        return stack.pop()

    def operations(self, a, b, operator):
        if operator == '+':
            return a + b
        elif operator == '-':
            return a - b
        elif operator == '*':
            return a * b
        return int(a / b) 

File: 150_evaluate_reverse_polish_notation/eval_rpn.rs
use std::collections::HashSet;

impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack: Vec<i32> = Vec::new();
        let operations: HashSet<&str> = ["+", "-", "*", "/"].iter().cloned().collect();

        for t in &tokens {
            if operations.contains(t.as_str()) {
                let b = stack.pop().unwrap();
                let a = stack.pop().unwrap();
                let res = match t.as_str() {
                    "+" => a + b,
                    "-" => a - b,
                    "*" => a * b,
                    "/" => a / b,
                    _ => unreachable!(),
                };
                stack.push(res);
            } else {
                stack.push(t.parse::<i32>().unwrap());
            }
        }
        return stack.last().copied().unwrap();
    }

}